### PR TITLE
Integrated Chat SDK Pre chat tests

### PR DESCRIPTION
**Pull request by gitanjuvs**
This PR includes the following chat scenarios,

1. ChatSDK.startChat() with preChatResponse should be part of session init payload (authenticated chat).
2. ChatSDK.getPreChatSurvey() should return the survey in JSON format.
3. ChatSDK.getPreChatSurvey() with parseToJSON set to true should return the survey as JSON format.
4. ChatSDK.getPreChatSurvey() with parseToJSON set to false should return the survey as string format.

### CODE REVIEW

File: playwright/integrations/unauthenticated-chat-with-prechat.spec.ts [hunk 1]
```diff
@@ -57,4 +57,142 @@ test.describe('UnauthenticatedChat @UnauthenticatedChatWithPrechat', () => {
         const { preChatResponse: preChatResponseData } = requestPostData;
         expect(optionalParams.preChatResponse).toEqual(preChatResponseData);
     });
+
+    test('ChatSDK.getPreChatSurvey() should return the survey in JSON format', async ({ page }) => {
+
+        await page.goto(testPage);
+
+        const optionalParams = {
```
Comment (xTEddie) [inline playwright/integrations/unauthenticated-chat-with-prechat.spec.ts:65]: We do not need to call `chatSDK.startChat()` & `chatSDK.endChat()`. We can simply validate the prechat survey after `chatSDK.initialize()`

You simply need to validate the `typeof preChatSurvey === 'object'`

File: playwright/integrations/unauthenticated-chat-with-prechat.spec.ts [hunk 2]
```diff
@@ -57,4 +57,142 @@ test.describe('UnauthenticatedChat @UnauthenticatedChatWithPrechat', () => {
...
+        const prechatSurveyBody = JSON.parse(preChatSurvey.body[2].id);
+        const prechatSurveyResponseBody = JSON.parse(optionalParams.preChatResponse.Survey);
+        expect(prechatSurveyBody.Name).toEqual(prechatSurveyResponseBody.Name);
+    });
+
+    test('ChatSDK.getPreChatSurvey() with parseToJSON set to false should return the survey as string format', async ({ page }) => {
+
+        await page.goto(testPage);
+
+        const optionalParams = {
```
Comment (xTEddie) [inline playwright/integrations/unauthenticated-chat-with-prechat.spec.ts:157]: We do not need to call `chatSDK.startChat()` & `chatSDK.endChat()`. We can simply validate the prechat survey after `chatSDK.initialize()`

You simply need to validate the `typeof preChatSurvey === 'string'`

File: playwright/integrations/unauthenticated-chat-with-prechat.spec.ts [hunk 3]
```diff
@@ -57,4 +57,142 @@ test.describe('UnauthenticatedChat @UnauthenticatedChatWithPrechat', () => {
...
+            }, { omnichannelConfig, optionalParams })
+        ]);
+
+        const { preChatSurvey } = runtimeContext;
+        const prechatSurveyBody = JSON.parse(preChatSurvey.body[2].id);
+        const prechatSurveyResponseBody = JSON.parse(optionalParams.preChatResponse.Survey);
+        expect(prechatSurveyBody.Name).toEqual(prechatSurveyResponseBody.Name);
+    });
+
+    test('ChatSDK.getPreChatSurvey() with parseToJSON set to true should return the survey as JSON format', async ({ page }) => {
```
Comment (xTEddie) [inline playwright/integrations/unauthenticated-chat-with-prechat.spec.ts:106]: `sessionInitRequest` & `sessionInitResponse` can all be removed

File: playwright/integrations/unauthenticated-chat-with-prechat.spec.ts (hunk 1 above)
Comment (gitanjuvs) [reply playwright/integrations/unauthenticated-chat-with-prechat.spec.ts]: Removed and added the validation.

File: playwright/integrations/unauthenticated-chat-with-prechat.spec.ts (hunk 3 above)
Comment (gitanjuvs) [reply playwright/integrations/unauthenticated-chat-with-prechat.spec.ts:83]: Removed

**Review by xTEddie — APPROVED** [review]

**Pull request merged**